src/System.Private.CoreLib/shared/System/Threading/Thread.cs

```diff
@@ -150,6 +155,30 @@ public static IPrincipal CurrentPrincipal
             }
         }
 
+        public static Thread CurrentThread => t_currentThread ?? InitializeCurrentThread();
+
+        public ExecutionContext ExecutionContext => ExecutionContext.Capture();
+
+        public string Name
+        {
+            get => _name;
+            set
+            {
+                if (Interlocked.CompareExchange(ref _name, value, null) != null)
+                {
+                    throw new InvalidOperationException(SR.InvalidOperation_WriteOnce);
+                }
+
+                ThreadNameChanged();
+            }
+        }
+
+        internal SynchronizationContext SynchronizationContext
```

Comment:
Maybe get rid of this property and change the places that need it to work directly on `_synchronizationContext`, so that the patterns for `_synchronizationContext` and `_executionContext` are the same

Reply:
done